### DISCUSSION TITLE
Using https scheme for redirect URI when working with a native app

### DISCUSSION THREAD

**Discussion by vin047** [zitadel/zitadel#4087] [General]
Zitadel console currently states:

> Redirect URIs must begin with your own protocol, http://127.0.0.1, http://[::1] or http://localhost.

And indeed, trying to add a URL with https results in a warning in the console.

According to https://www.oauth.com/oauth2-servers/oauth-native-apps/redirect-urls-for-native-apps/, it's best to use HTTPS URL matching when working with native apps, as custom schemes can be registered by other applications and which app the system chooses to launch is random. 

I'm aware that I can just turn on `Development Mode` but this feels like something that should be supported out of the box, at least for native apps.

**Comment by fforootd** [comment]
True this is worth discussing. @livio-a whats your take on this?

**Reply by livio-a** [reply]
@vin047 you're absolutely right. HTTPS should be allowed for any type and I found the original recommendation of using it for claimed uri redirects in the [OAuth 2.0 for Native Apps (RFC8252)](https://datatracker.ietf.org/doc/html/rfc8252#section-7.2)

I'll open an issue for this. In the meantime, you'll have to use the `Development Mode` workaround.
